Mathlib/SetTheory/Ordinal/Notation.lean

```diff
@@ -0,0 +1,1384 @@
+    · exact ⟨rfl, inferInstance⟩
+    · have := opow_pos (repr a') omega_pos
+      refine'
+        ⟨mul_isLimit this omega_isLimit, fun i =>
+          ⟨this, _, fun H => @NF.oadd_zero _ _ (iha.2 H.fst)⟩, exists_lt_mul_omega'⟩
+      rw [← mul_succ, ← nat_cast_succ, Ordinal.mul_lt_mul_iff_left this]
+      apply nat_lt_omega
+    · have := opow_pos (repr a') omega_pos
+      refine'
+        ⟨add_isLimit _ (mul_isLimit this omega_isLimit), fun i => ⟨this, _, _⟩,
+          exists_lt_add exists_lt_mul_omega'⟩
+      · rw [← mul_succ, ← nat_cast_succ, Ordinal.mul_lt_mul_iff_left this]
+        apply nat_lt_omega
+      · refine' fun H => H.fst.oadd _ (NF.below_of_lt' _ (@NF.oadd_zero _ _ (iha.2 H.fst)))
+        rw [repr, ← zero_def, repr, add_zero, iha.1, opow_succ, Ordinal.mul_lt_mul_iff_left this]
+        apply nat_lt_omega
+    · rcases iha with ⟨h1, h2, h3⟩
+      refine' ⟨opow_isLimit one_lt_omega h1, fun i => _, exists_lt_omega_opow' one_lt_omega h1 h3⟩
+      obtain ⟨h4, h5, h6⟩ := h2 i
+      exact ⟨h4, h5, fun H => @NF.oadd_zero _ _ (h6 H.fst)⟩
+    · rcases iha with ⟨h1, h2, h3⟩
+      refine'
+        ⟨add_isLimit _ (opow_isLimit one_lt_omega h1), fun i => _,
+          exists_lt_add (exists_lt_omega_opow' one_lt_omega h1 h3)⟩
+      obtain ⟨h4, h5, h6⟩ := h2 i
+      refine' ⟨h4, h5, fun H => H.fst.oadd _ (NF.below_of_lt' _ (@NF.oadd_zero _ _ (h6 H.fst)))⟩
+      rwa [repr, ← zero_def, repr, add_zero, PNat.one_coe, Nat.cast_one, mul_one,
+        opow_lt_opow_iff_right one_lt_omega]
+  · refine'
+      ⟨by rw [repr, ihb.1, add_succ, repr], fun H => H.fst.oadd _ (NF.below_of_lt' _ (ihb.2 H.snd))⟩
+    have := H.snd'.repr_lt
+    rw [ihb.1] at this
+    exact (lt_succ _).trans this
+  · rcases ihb with ⟨h1, h2, h3⟩
+    simp only [repr]
+    exact
+      ⟨Ordinal.add_isLimit _ h1, fun i =>
+        ⟨oadd_lt_oadd_3 (h2 i).1, oadd_lt_oadd_3 (h2 i).2.1, fun H =>
+          H.fst.oadd _ (NF.below_of_lt' (lt_trans (h2 i).2.1 H.snd'.repr_lt) ((h2 i).2.2 H.snd))⟩,
+        exists_lt_add h3⟩
+#align onote.fundamental_sequence_has_prop ONote.fundamentalSequence_has_prop
+
+/-- The fast growing hierarchy for ordinal notations `< ε₀`. This is a sequence of
+functions `ℕ → ℕ` indexed by ordinals, with the definition:
+* `f_0(n) = n + 1`
+* `f_(α+1)(n) = f_α^[n](n)`
+* `f_α(n) = f_(α[n])(n)` where `α` is a limit ordinal
+   and `α[i]` is the fundamental sequence converging to `α` -/
+def fastGrowing : ONote → ℕ → ℕ
+  | o =>
+    match fundamentalSequence o, fundamentalSequence_has_prop o with
+    | Sum.inl none, _ => Nat.succ
+    | Sum.inl (some a), h =>
+      have : a < o := by rw [lt_def, h.1]; apply lt_succ
+      fun i => (fastGrowing a^[i]) i
+    | Sum.inr f, h => fun i =>
+      have : f i < o := (h.2.1 i).2.1
+      fastGrowing (f i) i
+  termination_by fastGrowing o => o
+#align onote.fast_growing ONote.fastGrowing
+
+-- Porting note: `this` in the definition of `fastGrowing` gets lint.
```

Comment:
Sorry, don't understand "gets lint". I assume this is failing one of the linters?

Reply:
There are still comments that say "gets lint". Could we fix those prior to merging?